naga/src/valid/function.rs

```diff
@@ -613,8 +728,27 @@ impl super::Validator {
                     stages &= super::ShaderStages::FRAGMENT;
                     finished = true;
                 }
-                S::Barrier(_) => {
+                S::Barrier(barrier) => {
                     stages &= super::ShaderStages::COMPUTE;
+                    if barrier.contains(crate::Barrier::SUB_GROUP) {
```

Comment:
HLSL doesn't have a subgroup barrier.
This would have to be gated by a capability (and a wgpu feature).

Reply:
Putting in a vote for having a subgroup barrier when available, as it's useful for sorting (particular the WLMS algorithm).

Reply:
This PR already adds support for the subgroup barrier.
It's just that they have to be gated behind another capability (and wgpu feature) since HLSL doesn't have it.